File: modules/live-activity/src/LiveActivityModule.ts
import { requireOptionalNativeModule } from 'expo-modules-core';

// Native module interface (implemented in ios/LiveActivityModule.swift).
// The Swift layer receives primitive args, so the native signatures are flattened
// (e.g. `startWorkActivity(clockInAt, targetLeaveAt)`), not object-shaped.
export type NativeLiveActivityModule = {
  areActivitiesEnabled(): boolean;
  startWorkActivity(clockInAt: string, targetLeaveAt: string): Promise<string | null>;
  endWorkActivity(): Promise<void>;
  updateWorkActivity(targetLeaveAt: string): Promise<void>;
  hasActiveWorkActivity(): Promise<boolean>;
};

// `requireOptionalNativeModule` returns `null` (instead of throwing) when the native
// module is not present — e.g. Android, web, or before a native build has been made.
export default requireOptionalNativeModule<NativeLiveActivityModule>('LiveActivityModule');

File: modules/live-activity/ios/LiveActivityModule.swift
import ActivityKit
import ExpoModulesCore

public class LiveActivityModule: Module {
  public func definition() -> ModuleDefinition {
    Name("LiveActivityModule")

    Function("areActivitiesEnabled") { () -> Bool in
      if #available(iOS 16.2, *) {
        return ActivityAuthorizationInfo().areActivitiesEnabled
      }
      return false
    }

    // The following are stubs — real ActivityKit wiring lands in Phase 3 Task 5.
    AsyncFunction("startWorkActivity") { (clockInAt: String, targetLeaveAt: String) -> String? in
      return nil
    }

    AsyncFunction("endWorkActivity") { () in
    }

    AsyncFunction("updateWorkActivity") { (targetLeaveAt: String) in
    }

    AsyncFunction("hasActiveWorkActivity") { () -> Bool in
      return false
    }
  }
}
